src/gmp/models/report/parser.js

```diff
@@ -52,89 +52,60 @@ const emptyCollectionList = filter => {
-  return cert;
-};
-
 export const parseTlsCertificates = (report, filter) => {
-  const {host: hosts, ssl_certs, results} = report;
+  const {ssl_certs, tls_certificates, results} = report;
 
-  if (!isDefined(ssl_certs) || !isReportWithDetails(results)) {
+  if (
+    !isDefined(ssl_certs) ||
+    !isDefined(tls_certificates) ||
+    !isReportWithDetails(results)
+  ) {
     return emptyCollectionList(filter);
   }
 
   const {count: full_count} = ssl_certs;
 
-  let certs_array = [];
-
-  forEach(hosts, host => {
-    const host_certs = {};
-    let hostname;
-
-    forEach(host.detail, detail => {
-      const {name = '', value = ''} = detail;
-
-      if (name.startsWith('SSLInfo')) {
-        const [port, fingerprint] = value.split('::');
-
-        const cert = getTlsCertificate(host_certs, fingerprint);
-
-        cert.ip = host.ip;
-
-        cert.addPort(port);
-      } else if (name.startsWith('SSLDetails')) {
-        const [, fingerprint] = name.split(':');
-
-        const cert = getTlsCertificate(host_certs, fingerprint);
-
-        value.split('|').reduce((c, v) => {
-          let [key, val] = v.split(':');
-          if (key === 'notAfter' || key === 'notBefore') {
-            val = parseDate(val);
-          }
-          c[key.toLowerCase()] = val;
-          return c;
-        }, cert);
-
-        cert.details = value;
-      } else if (name.startsWith('Cert')) {
-        const [, fingerprint] = name.split(':');
-
-        const cert = getTlsCertificate(host_certs, fingerprint);
-
-        // currently cert data starts with x509:
-        // not sure if there are other types of certs
-        // therefore keep original data
-
-        cert._data = value;
-
-        if (value.includes(':')) {
-          const [, data] = value.split(':');
-          cert.data = data;
-        } else {
-          cert.data = value;
-        }
-      } else if (name === 'hostname') {
-        // collect hostnames
-        hostname = value;
-      }
+  const certs_array = [];
+
+  forEach(tls_certificates.tls_certificate, tls_cert => {
+    const {
+      name,
+      certificate,
+      sha256_fingerprint,
+      md5_fingerprint,
+      valid,
+      activation_time,
+      expiration_time,
+      subject_dn,
+      issuer_dn,
+      serial,
+      host,
+      ports,
+    } = tls_cert;
+
+    const cert = ReportTLSCertificate.fromElement({fingerprint: name});
+    cert.data = isDefined(certificate) ? certificate.__text : undefined;
+    cert.sha256Fingerprint = sha256_fingerprint;
+    cert.md5Fingerprint = md5_fingerprint;
+    cert.activationTime =
+      activation_time === 'undefined' || activation_time === 'unlimited'
+        ? undefined
+        : parseDate(activation_time);
+    cert.expirationTime =
+      expiration_time === 'undefined' || expiration_time === 'unlimited'
+        ? undefined
+        : parseDate(expiration_time);
+    cert.valid = parseBoolean(valid);
+    cert.subject_dn = subject_dn;
+    cert.issuer_dn = issuer_dn;
```

Comment:
Shouldn't both be camelCase too?

Reply:
Agree. Asset TLS certificate names updated as well since both components share `TlsCertificateDetails`.